File: src/ExpoAlipayModule.ts
import { NativeModule, requireNativeModule } from 'expo';

import { AuthOptions, ExpoAlipayModuleEvents, PayOptions } from './ExpoAlipay.types';

declare class ExpoAlipayModule extends NativeModule<ExpoAlipayModuleEvents> {
  /**
   * 获取支付宝 SDK 版本号。
   */
  getVersion(): Promise<string>;

  /**
   * 注册支付宝应用（Android 防黑产场景使用）。
   * @param appId 支付宝开放平台应用 ID。
   */
  registerApp(appId: string): Promise<boolean>;

  /**
   * 设置沙箱模式（仅 Android）。
   * @param mode "sandbox" | "online" | "pre_sandbox"
   */
  setSandboxMode(mode: 'sandbox' | 'online' | 'pre_sandbox'): Promise<void>;

  /**
   * 支付宝支付。
   * @param options 支付参数。
   * @returns 支付调用结果。
   */
  pay(options: PayOptions): Promise<any>;

  /**
   * 支付宝授权登录。
   * @param options 授权参数。
   * @returns 授权调用结果。
   */
  auth(options: AuthOptions): Promise<any>;

  /**
   * 开启支付宝 SDK 日志（仅 iOS）。
   */
  startLog(): Promise<void>;

  /**
   * 关闭支付宝 SDK 日志（仅 iOS）。
   */
  stopLog(): Promise<void>;
}

// This call loads the native module object from the JSI.
export default requireNativeModule<ExpoAlipayModule>('ExpoAlipay');

File: android/src/main/java/expo/modules/alipay/ExpoAlipayModule.kt
package expo.modules.alipay

import android.util.Log
import com.alipay.sdk.app.AlipayApi
import com.alipay.sdk.app.AuthTask
import com.alipay.sdk.app.EnvUtils
import com.alipay.sdk.app.PayTask
import expo.modules.kotlin.Promise
import expo.modules.kotlin.exception.CodedException
import expo.modules.kotlin.exception.toCodedException
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.net.URL

class ExpoAlipayModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("ExpoAlipay")

        Events("onPayResult", "onAuthResult")

        AsyncFunction("registerApp") { appId: String ->
            appContext.reactContext?.let {
                AlipayApi.registerApp(it, appId)
            }
        }

        AsyncFunction("getVersion") {
            val payTask = PayTask(appContext.currentActivity)
            return@AsyncFunction payTask.version
        }

        AsyncFunction("setSandboxMode") { mode: String ->

            when (mode) {
                "sandbox" -> {
                    EnvUtils.setEnv(EnvUtils.EnvEnum.SANDBOX)
                }

                "online" -> {
                    EnvUtils.setEnv(EnvUtils.EnvEnum.ONLINE)
                }

                "pre_sandbox" -> {
                    EnvUtils.setEnv(EnvUtils.EnvEnum.PRE_SANDBOX)
                }
            }
        }

        AsyncFunction("auth") { options: AuthOptions, promise: Promise ->
            val runnable = Runnable {
                try {
                    val alipay = AuthTask(appContext.currentActivity)
                    val result = alipay.authV2(options.authInfo, true)
                    promise.resolve(result)
                    sendEvent("onAuthResult", result)
                } catch (e: Exception) {
                    promise.reject(e.toCodedException())
                }
            }
            Thread(runnable).start()
        }

        AsyncFunction("pay") { options: PayOptions, promise: Promise ->
            val runnable = Runnable {
                try {
                    val alipay = PayTask(appContext.currentActivity)
                    val result = alipay.payV2(options.orderInfo, true)
                    promise.resolve(result)
                    sendEvent("onPayResult", result)
                } catch (e: Exception) {
                    promise.reject(e.toCodedException())
                }
            }

            Thread(runnable).start()
        }


        AsyncFunction("startLog") {
            Log.d("ExpoAlipay", "startLog called (no-op on Android)")
        }

        AsyncFunction("stopLog") {
            Log.d("ExpoAlipay", "stopLog called (no-op on Android)")
        }
    }
}
